### PR TITLE
attempt to support go module

### PR DESCRIPTION
**Pull request by thinkerou**
Ensure old version run right and then attempt to support go module.
please @javierprovecho @appleboy help review, thanks!

### CODE REVIEW

**Comment by thinkerou** [comment]
@appleboy can you help see the pull request why waiting? Because `.travis.yml` have some syntax error? thanks!

File: Makefile [hunk 1]
```diff
@@ -60,3 +60,9 @@ misspell:
 		go get -u github.com/client9/misspell/cmd/misspell; \
 	fi
 	misspell -w $(GOFILES)
+
+.PHONY: tools
+tools:
+	go install github.com/golang/lint/golint; \
```
Comment (appleboy) [inline Makefile]: change golint path to `golang.org/x/lint/golint`

Comment (thinkerou) [reply Makefile]: thanks, done.

**Comment by rarkins** [comment]
Here are some other `go.mod` updates you could make: https://github.com/renovate-mirrors/gin/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc

image

**Comment by rarkins** [comment]
@thinkerou Renovate Bot is something I wrote that can be installed into any repo, so if you wanted the PRs *here* then you could [install it](https://github.com/apps/renovate). 

Otherwise, I have set up the "mirror" that will keep in sync each hour or two with this parent repo and always maintain a list of outdated dependencies in PRs for you to see. FYI, I used this repository as one of my test repositories when testing support for go modules in the bot.

If you don't want to install the bot but would like to upgrade some of the modules it suggests, you can do this:

1. View the list of branches in the mirror: https://github.com/renovate-mirrors/gin/branches/active
2. Click on the branch you want
3. Click on "Compare"
4. Click on "Create Pull Request"

GitHub should then let you raise your own PR into this repository from the branch in the mirror.

**Comment by thinkerou** [comment]
@rarkins thanks, got it. Because gin still mainly use `govendor`, we need to upgrade vendor.json and then upgrade go.mod/go.sum. I will fix the issue tonight.

**Comment by kidlj** [comment]
`godep save .` fails because of these blank imports.

```
$ godep save
godep: Package (github.com/client9/misspell/cmd/misspell) not found
```

**Pull request merged**